### PR TITLE
Update to Sentry SDK 6.3.0. Provide flows for users app context.

### LINKED ISSUE #7057: Provide asynchronous updates data for crash logging
Because of reasons explained in the description of https://github.com/Automattic/Automattic-Tracks-Android/pull/137, we have to satisfy a new interface by providing data flows for user and application context details.

### PR DESCRIPTION
**Pull request by wzieba**
Blocked by https://github.com/Automattic/Automattic-Tracks-Android/pull/137

Closes: #7057

### Description

This PR adjusts `WCCrashLoggingProvider` to satisfy changes introduced in https://github.com/Automattic/Automattic-Tracks-Android/pull/137.

### Testing instructions

Apply PATCH, and run the app. In Sentry dashboard, see that there's a new issue with correct site URL, site id and username.

<details>
<summary> PATCH </summary>

```PATCH
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt	(revision b4a24bcd8cb2a2e485d279cc9a7285fdd38cbcda)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt	(date 1659431969168)
@@ -275,6 +275,10 @@
         if (!BuildConfig.DEBUG) {
             checkForAppUpdates()
         }
+
+        crashLogging.sendReport(
+            message = "This is just a test. Remove me."
+        )
     }
 
     override fun hideProgressDialog() {
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt	(revision b4a24bcd8cb2a2e485d279cc9a7285fdd38cbcda)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt	(date 1659431692898)
@@ -80,7 +80,8 @@
     override val user: Flow<CrashLoggingUser?> = crashLoggingUser
 
     override fun crashLoggingEnabled(): Boolean {
-        return appPrefs.isCrashReportingEnabled()
+        return true
+//        return appPrefs.isCrashReportingEnabled()
     }
 
     override fun extraKnownKeys(): List<ExtraKnownKey> {

```
</details>

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/build.gradle [hunk 1]
```diff
@@ -13,6 +13,15 @@ plugins {
     id 'com.google.gms.google-services'
 }
 
+sentry {
+    tracingInstrumentation {
+        enabled = false
+    }
+    autoInstallation {
+        enabled = false
+    }
```
Comment (wzieba) [inline WooCommerce/build.gradle:22]: Disables newly added default behavior of Sentry SDK. We initialize the services we need manually and only those we really need.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt [hunk 2]
```diff
@@ -6,22 +6,62 @@ import com.automattic.android.tracks.crashlogging.EventLevel
...
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.model.AccountModel
 import org.wordpress.android.fluxc.store.AccountStore
+import org.wordpress.android.fluxc.store.AccountStore.OnAccountChanged
 import java.util.Locale
 import javax.inject.Inject
+import javax.inject.Singleton
 
+@Singleton
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt:28]: As `WCCrashLoggingDataProvider` starts to have a state of `crashLoggingUser`, we have to make it a singleton.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProviderTest.kt [hunk 3]
```diff
@@ -188,6 +211,14 @@ class WCCrashLoggingDataProviderTest {
         assertThat(sut.releaseName).isEqualTo(DEBUG_RELEASE_NAME)
     }
 
+    private fun softlyAssertUser(user: CrashLoggingUser?) {
+        SoftAssertions().apply {
```
Comment (atorresveiga) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProviderTest.kt:215]: Nice!

**Review by atorresveiga — APPROVED** [review]
Great work @wzieba! I have tested the changes and everything is working as expected.🎉

Screen Shot 2022-08-03 at 18 37 11

**Pull request merged**